File: 01_big_O_notation/05_rules/03_product_rule/product_rule.js
/*
 f(n) = O(p(n))
                -> O(p(n)q(n))
 g(n) = O(q(n))
*/

function product_fg(n) {
    let accum = 0

    // f
    for (let i = 0; i < n; i ++) {
        accum += 1

        // g
        for (let j = 0; j < n; j ++) {
            accum += 1
        }
    }

    return accum;
}

n = 5
console.log(product_fg(n))

File: 01_big_O_notation/05_rules/03_product_rule/product_rule.py
"""
 f(n) = O(p(n))
                -> O(p(n)q(n))
 g(n) = O(q(n))
"""


def product_fg(n):
    accum = 0

    # f
    for i in range(n):
        accum += 1

        # g
        for j in range(n):
            accum += 1

    return accum


if __name__ == '__main__':
    n = 5
    print(product_fg(n))
